m2cgen/ast.py

```diff
@@ -63,6 +81,12 @@ def __str__(self):
         args = ",".join([str(self.expr), "to_reuse=" + str(self.to_reuse)])
         return "AbsExpr(" + args + ")"
 
+    def __eq__(self, other):
+        return type(other) is AbsExpr and self.expr == other.expr
```

Comment:
It bothers me semantically that we don't check the `to_reuse` flag for equality too.

Reply:
I believe it is OK to not carry about unimportant attributes of objects during comparison. Adding `to_reuse` flag here will eliminate practically the whole point of this PR.